applications/dashboard/settings/class.hooks.php

```diff
@@ -135,6 +135,14 @@ public function base_render_before($Sender) {
 
         // Allow global translation of TagHint
         $Sender->addDefinition("TagHint", t("TagHint", "Start to type..."));
+
+        // Add user's viewable roles to gdn.meta if user is logged in.
+        if (!array_key_exists('Roles', $Sender->getDefinitions())) {
```

Comment:
To check if a definition exists, I would add a controller method called `hasDefinition($term)` instead of `getDefinitions()`.
IMO, the resulting condition would read better like `if (!$this->hasDefinition('Roles')) {`